src/cli/commands/config.js

```diff
@@ -16,8 +16,15 @@ export const {run, setFlags} = buildSubCommands('config', {
       return false;
     }
 
-    const [key, val = true] = args;
+    const [key, val] = ((key, val = true) => {
+      if (typeof val === 'string' && (/^(true|false)$/i).test(val)) {
+        val = (val.toLowerCase() === 'true');
+      }
+
+      return [key, val];
+    })(...args);
```

Comment:
```javascript
const key = args[0];
let val = args[1];
```

You'll loose the default `true` value this way from the `val` assignment (I didn't want to introduce a regression). It *could* be replaced with `let val = args[1] || true` but it's not the same as a default value (falsy values don't work, most importantly `0`), hence IIFE.

```javascript
if (key === 'true' ...) {
   val = val === 'true';
}
```

See my previous comment.

> We don't really need to handle the alternate casing of `true`.

What about calling `config set` without the value? AFAIK, NPM defaults to an empty string.